wrapper/src/main/java/software/amazon/jdbc/MonitorPluginService.java

```diff
@@ -0,0 +1,795 @@
+    if (this.exceptionHandler != null) {
+      return this.exceptionHandler.isLoginException(throwable, targetDriverDialect);
+    }
+    return this.exceptionManager.isLoginException(this.dbDialect, throwable, targetDriverDialect);
+  }
+
+  @Override
+  public boolean isLoginException(final String sqlState) {
+    if (this.exceptionHandler != null) {
+      return this.exceptionHandler.isLoginException(sqlState);
+    }
+    return this.exceptionManager.isLoginException(this.dbDialect, sqlState);
+  }
+
+  @Override
+  public Dialect getDialect() {
+    return this.dbDialect;
+  }
+
+  @Override
+  public TargetDriverDialect getTargetDriverDialect() {
+    return this.targetDriverDialect;
+  }
+
+  @Override
+  public void updateDialect(final @NonNull Connection connection) {
+    // TODO: is it fine to just leave this empty? This method is called after connecting in DefaultConnectionPlugin but
+    //  the dialect passed to the constructor should already be up to date.
+    // do nothing.
+  }
+
+  @Override
+  public HostSpec identifyConnection(Connection connection) throws SQLException {
+    return this.getHostListProvider().identifyConnection(connection);
+  }
+
+  @Override
+  public void fillAliases(Connection connection, HostSpec hostSpec) throws SQLException {
+    if (hostSpec == null) {
+      return;
+    }
+
+    if (!hostSpec.getAliases().isEmpty()) {
+      LOGGER.finest(() -> Messages.get("PluginServiceImpl.nonEmptyAliases", new Object[] {hostSpec.getAliases()}));
+      return;
+    }
+
+    hostSpec.addAlias(hostSpec.asAlias());
+
+    // Add the host name and port, this host name is usually the internal IP address.
+    try (final Statement stmt = connection.createStatement()) {
+      try (final ResultSet rs = stmt.executeQuery(this.getDialect().getHostAliasQuery())) {
+        while (rs.next()) {
+          hostSpec.addAlias(rs.getString(1));
+        }
+      }
+    } catch (final SQLException sqlException) {
+      // log and ignore
+      LOGGER.finest(() -> Messages.get("PluginServiceImpl.failedToRetrieveHostPort"));
+    }
+
+    // Add the instance endpoint if the current connection is associated with a topology aware database cluster.
+    final HostSpec host = this.identifyConnection(connection);
+    if (host != null) {
+      hostSpec.addAlias(host.asAliases().toArray(new String[] {}));
+    }
+  }
+
+  @Override
+  public HostSpecBuilder getHostSpecBuilder() {
+    return new HostSpecBuilder(new HostAvailabilityStrategyFactory().create(this.props));
+  }
+
+  @Override
+  public Properties getProperties() {
+    return this.props;
+  }
+
+  public TelemetryFactory getTelemetryFactory() {
+    return this.pluginManager.getTelemetryFactory();
+  }
+
+  public String getTargetName() {
+    return this.pluginManager.getDefaultConnProvider().getTargetName();
+  }
+
+  @Override
+  public @NonNull SessionStateService getSessionStateService() {
+    return this.sessionStateService;
```

Comment:
Not sure about supporting session state. We don't have a current connection here. We don't need to take care after connection state.